### DISCUSSION TITLE
Tapo p125m

### DISCUSSION THREAD

**Discussion by requies** [petretiandrea/home-assistant-tapo-p100#616] [Ideas]
Can you please add monitor integration for P125m ?

Thank you

**Comment by madner81** [comment]
It would be great!

**Comment by giomurru** [comment]
P125M is detected only as a switch but energy monitoring is not. Please add support for it.

**Comment by giomurru** [comment]
I've added `"p125m"` to the list `SUPPORTED_DEVICE_AS_SWITCH_POWER_MONITOR` in `const.py` which I found on the  `custom_components/tapo/` directory inside the config directory of home assistant, then I restarted home assistant and all the 8 entities, including those regarding energy monitoring, seems to be working correctly.
I will let you know if my tapo plug will explode in the next days. LOL.

**Reply by requies** [reply]
Looks like it works perfectly.
I didn't think about playing with the config file.

Thank you

![tapo](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/2270251/ba95dcd4-cef0-4c9c-9b7a-8d8843fc1478)

**Reply by petretiandrea** [reply]
I will add
